compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -206,18 +220,49 @@ func (s *TestComplianceCachingSuite) TestScanReturnsCachedInventory() {
 	maxBackoff := 10 * time.Second
 	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
 	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, func(time.Duration) {})
+	validUntil := time.Now().Add(2 * time.Minute)
 
 	w := &inventoryWrap{
-		CacheValidUntil:      time.Now().Add(2 * time.Minute),
+		CacheValidUntil:      validUntil,
 		RetryBackoffDuration: 0,
 		CachedInventory:      "{\n  \"nodeId\": \"00000000-0000-0000-0000-000000000000\",\n  \"nodeName\": \"cachedNode\",\n  \"scanTime\": \"2023-11-11T11:11:11.382478080Z\",\n  \"components\": {\n    \"namespace\": \"unknown\"\n  },\n  \"notes\": [\n    \"LANGUAGE_CVES_UNAVAILABLE\"\n  ]\n}",
 	}
 	s.writeWrap(w, inventoryCachePath)
 
+	// Check directly returned results
 	actual, err := cs.Scan("testme")
-
 	s.NoError(err)
 	s.Equal(actual.NodeName, "cachedNode")
+
+	// Check cached results
+	wrap := s.readWrap(inventoryCachePath)
+	s.WithinDuration(validUntil, wrap.CacheValidUntil, 100*time.Millisecond, "Cache validity should not change when accessing a cached scan.")
```

Comment:
But why? Shouldn't `wrap.CacheValidUntil` be **exactly** equal to `validUntil`?

Reply:
It should be, correct. I compared with `s.Equal`, which lead to the test failing.
Digging into this, it looks like this is not the right way to go ([Testify issue](https://github.com/stretchr/testify/issues/1010)).
So what I have done now is set the delta to 0. This succeeds, as the last answer in that issue also indicates.

This is how the failing test with `s.Equal` looks:
```
Not equal: 
        	            	expected: time.Date(2023, time.March, 9, 15, 17, 43, 728595000, time.Local)
        	            	actual  : time.Date(2023, time.March, 9, 15, 17, 43, 728595000, time.Local)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(time.Time) 2023-03-09 15:17:43.728595 +0100 CET m=+120.034744843
        	            	+(time.Time) 2023-03-09 15:17:43.728595 +0100 CET
```

Reply:
Good job here